modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/matrix/stats/RunningStats.java

```diff
@@ -287,14 +287,21 @@ private void mergeCovariance(final RunningStats other, final Map<String, Double>
             // merge covariances
             if (covariances.containsKey(fieldName)) {
                 HashMap<String, Double> cFieldVals = covariances.get(fieldName);
-                for (String cFieldName : cFieldVals.keySet()) {
-                    newVal = cFieldVals.get(cFieldName);
-                    if (other.covariances.containsKey(fieldName) && other.covariances.get(fieldName).containsKey(cFieldName)) {
-                        newVal += other.covariances.get(fieldName).get(cFieldName) + f * dR * deltas.get(cFieldName);
+                for (Map.Entry<String, Double> stringDoubleEntry : cFieldVals.entrySet()) {
```

Comment:
This can have a better name like `cFieldVal` which since it's shorter hopefully means you can revert the line breaks making this change easier to inspect.